packages/react-menu/src/MenuItem/MenuItem.types.ts

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { ComponentProps, ShorthandProps } from '@fluentui/react-utils';
+
+export interface MenuItemProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Icon slot rendered before children content
+   */
+  icon?: ShorthandProps;
+}
+
+export interface MenuItemState extends MenuItemProps {}
```

Comment:
```suggestion
export interface MenuItemState extends MenuItemProps {
   icon?: TYPE
}
```

As we discussed offline, this should have a different type

Reply:
`ref` should be also typed

Reply:
typed to the same output as `useMergedRefs`